main.go

```diff
@@ -106,11 +107,23 @@ func main() {
 	http.Handle("/live", handlers.NewLivenessHandler())
 	http.Handle("/metrics", promhttp.HandlerFor(reg, promhttp.HandlerOpts{}))
 
-	log.Info("Starting http handler...")
-	address := fmt.Sprintf(":%s", Options.ListenPort)
+	httpsListen := func(port string) {
+		log.Infof("Starting https handler on %s...", port)
+		log.Fatal(http.ListenAndServeTLS(fmt.Sprintf(":%s", port), Options.HTTPSCertFile, Options.HTTPSKeyFile, nil))
+	}
+
+	httpListen := func(port string) {
+		log.Infof("Starting http handler on %s...", port)
+		log.Fatal(http.ListenAndServe(fmt.Sprintf(":%s", port), nil))
+	}
+
 	if Options.HTTPSKeyFile != "" && Options.HTTPSCertFile != "" {
-		log.Fatal(http.ListenAndServeTLS(address, Options.HTTPSCertFile, Options.HTTPSKeyFile, nil))
+		go httpsListen(Options.ListenPort)
+		if Options.HTTPListenPort != "" {
+			go httpListen(Options.HTTPListenPort)
+		}
+		select {}
```

Comment:
Right, this would not wait for goroutines to complete.

I don't think that affects anything, but lets keep it on hold until we confirm this doesn't affect any functionality

Reply:
Would it also be relevant to take into account if one server errors out? What would happen then?

Reply:
>if one server errors out

both goroutines will be terminated (ungracefully atm)

Reply:
Maybe we could use a [channel](https://gobyexample.com/channels) to listen for errors and terminate the servers if an error is received? Something like this?
```
         errs := make(chan error)
         httpListen := func(port string) {
		log.Infof("Starting http handler on %s...", port)
		if err := http.ListenAndServe(fmt.Sprintf(":%s", port), nil); err != nil {
                     errs <- err
                 }
         }
...
         select {
         case err := <-errs:
             log.Printf("Could not start serving service due to (error: %s)", err)
             // shut down servers
         }
```